File: 221116/MatchingTable.java
public class MatchingTable {
    public int solution(int participants, int number1, int number2) {
        int startRound = 1;
        int[] roundAndNumbers = new int[]{startRound, number1, number2};

        while (checkIfHaveToExecute(roundAndNumbers[1], roundAndNumbers[2])) {
            roundAndNumbers = executeRoundMatch(roundAndNumbers);
        }

        return roundAndNumbers[0];
    }

    public int[] executeRoundMatch(int[] roundAndNumbers) {
        int round = roundAndNumbers[0] + 1;

        if (roundAndNumbers[1] % 2 == 1) {
            roundAndNumbers[1] += 1;
        }
        if (roundAndNumbers[2] % 2 == 1) {
            roundAndNumbers[2] += 1;
        }

        return new int[]{
            round,
            roundAndNumbers[1] / 2,
            roundAndNumbers[2] / 2
        };
    }

    public boolean checkIfHaveToExecute(int number1, int number2) {
        if (Math.abs(number1 - number2) == 1
            && ((number1 > number2 && number1 % 2 == 0)
            || (number2 > number1 && number2 % 2 == 0))) {
            return false;
        }
        return true;
    }
}

File: solutions/Previous/221117/matchingTable.js
// export default function solution(participants, number1, number2) {
  
// }

// export function mapTest() {
//   const array = Array(8).fill(0).map((_, index) => index + 1);
//   console.log(array);

//   const pair = [];
//   const array1 = [];
//   const array2 = [];

//   array.forEach((index) => {
//     console.log(index);
//     pair.push(index);
//     console.log(pair);
//     if (pair.length === 2) {
//       array1.push(pair);
//       pair.splice(0);
//     }
//   });

//   console.log(array1[0]);
// }

export default function solution(n, a, b) {
  let answer = 1;

  while (Math.pow(2, answer) < n) {
    if (Math.abs(a - b) === 1 && Math.min(a, b) % 2 === 1) {
      break;
    }

    a = Math.ceil(a / 2);
    b = Math.ceil(b / 2);

    answer += 1;
  }

  return answer;
}
